runtime/src/main/cpp/Worker.cpp

```diff
@@ -374,27 +504,46 @@ void* workerRoutine(void* argument) {
     ObjHolder resultHolder;
     while (true) {
       Job job = worker->getJob();
-      if (job.function == nullptr) {
+      if (job.kind == JOB_TERMINATE) {
+        if (job.terminationRequest.waitDelayed) {
+          if (worker->waitDelayed()) {
+            worker->putJob(job, false);
+            continue;
+          }
+        }
         // Termination request, notify the future.
-        job.future->storeResultUnlocked(nullptr, true);
+        job.terminationRequest.future->storeResultUnlocked(nullptr, true);
         theState()->removeWorkerUnlocked(worker->id());
         break;
       }
-      KRef argument = AdoptStablePointer(job.argument, argumentHolder.slot());
+      if (job.kind == JOB_EXECUTE_AFTER) {
+         ObjHolder operationHolder, dummyHolder;
+         KRef obj = DerefStablePointer(job.executeAfter.operation, operationHolder.slot());
+         try {
+           WorkerLaunchpad(obj, dummyHolder.slot());
+         } catch (ExceptionObjHolder& e) {
+           if (worker->errorReporting())
+             ReportUnhandledException(e.obj());
+         }
+         DisposeStablePointer(job.executeAfter.operation);
```

Comment:
Why not perform this immediately after `DerefStablePointer`?

Reply:
Not sure if it's too important. It is important in other places where ownership is transferred, but here we kind of own (or share) this lambda anyway.